presto-main/src/main/java/com/facebook/presto/operator/scalar/ArraySliceFunction.java

```diff
@@ -13,61 +13,28 @@
  */
 package com.facebook.presto.operator.scalar;
 
-import com.facebook.presto.metadata.FunctionRegistry;
-import com.facebook.presto.metadata.SqlScalarFunction;
+import com.facebook.presto.operator.Description;
 import com.facebook.presto.spi.block.Block;
 import com.facebook.presto.spi.block.BlockBuilderStatus;
 import com.facebook.presto.spi.type.Type;
-import com.facebook.presto.spi.type.TypeManager;
-import com.google.common.collect.ImmutableList;
+import com.facebook.presto.type.SqlType;
 
-import java.lang.invoke.MethodHandle;
-import java.util.Map;
-
-import static com.facebook.presto.metadata.Signature.typeParameter;
 import static com.facebook.presto.spi.StandardErrorCode.INVALID_FUNCTION_ARGUMENT;
+import static com.facebook.presto.spi.type.StandardTypes.BIGINT;
 import static com.facebook.presto.util.Failures.checkCondition;
-import static com.facebook.presto.util.Reflection.methodHandle;
 
+@ScalarFunction("slice")
+@Description("Subsets an array given an offset (1-indexed) and length")
 public final class ArraySliceFunction
-        extends SqlScalarFunction
 {
-    public static final ArraySliceFunction ARRAY_SLICE_FUNCTION = new ArraySliceFunction();
-    private static final String FUNCTION_NAME = "slice";
-    private static final MethodHandle METHOD_HANDLE = methodHandle(ArraySliceFunction.class, "slice", Type.class, Block.class, long.class, long.class);
-
-    public ArraySliceFunction()
-    {
-        super(FUNCTION_NAME, ImmutableList.of(typeParameter("E")), "array(E)", ImmutableList.of("array(E)", "bigint", "bigint"));
-    }
-
-    @Override
-    public boolean isHidden()
-    {
-        return false;
-    }
-
-    @Override
-    public boolean isDeterministic()
-    {
-        return true;
-    }
-
-    @Override
-    public String getDescription()
-    {
-        return "Subsets an array given an offset (1-indexed) and length";
-    }
-
-    @Override
-    public ScalarFunctionImplementation specialize(Map<String, Type> types, int arity, TypeManager typeManager, FunctionRegistry functionRegistry)
-    {
-        Type type = types.get("E");
-        MethodHandle methodHandle = METHOD_HANDLE.bindTo(type);
-        return new ScalarFunctionImplementation(false, ImmutableList.of(false, false, false), methodHandle, isDeterministic());
-    }
-
-    public static Block slice(Type type, Block array, long fromIndex, long length)
+    private ArraySliceFunction() {}
+
+    @TypeParameter("E")
+    @SqlType("array(E)")
+    public static Block slice(@TypeParameter("E") Type type,
```

Comment:
When function parameters are declared on multiple lines, always put the first one on its own line.